snakemake/common/plugin.py

```diff
@@ -0,0 +1,132 @@
+__author__ = "Hielke Walinga"
+__copyright__ = "Copyright 2022, Hielke Walinga"
+__email__ = "hielkewalinga@gmail.com"
+__license__ = "MIT"
+
+import importlib
+import pkgutil
+import sys
+from os import path
+
+# TODO: When 3.9 deprecated, import from importlib.metadata
+from snakemake.common.future import packages_distributions
+from snakemake.exceptions import PluginException
+from snakemake.logging import logger
+from snakemake.utils import min_version
+
+cached_packages_distributions = packages_distributions()
+
+
+def find_plugins(prefix):
+    """Iterator to find all plugins by their prefix of the package name."""
+    plugins = []
+    for finder, name, ispkg in pkgutil.iter_modules():
+        # Check if the *package name* of the module
+        # start with the correct prefix.
+        try:
+            packages = cached_packages_distributions[name]
+        except KeyError:
+            continue
+        if packages[0].startswith(prefix):
```

Comment:
What if there is more than one item in packages? Can that happen? What would that mean?

Reply:
Would it make sense to throw an error if there is more than one?

Reply:
If you install a package, the package name does not have to be equal to the module that the package installs. You can install multiple packages that provide a module with the same name. So packages_distributions lists all packages that provide that module. Which is always one, but in a weird situation, it can be more. But if there are more, the first one is always used. Also by `import` (I tested that). But yes, it would make sense to throw an error. I will add that.